common/types/timestamp.go

```diff
@@ -272,12 +274,33 @@ func timestampGetMillisecondsWithTz(t time.Time, tz ref.Val) ref.Val {
 
 func timeZone(tz ref.Val, visitor timestampVisitor) timestampVisitor {
 	return func(t time.Time) ref.Val {
+		var err error
+		var offset int
 		if StringType != tz.Type() {
 			return ValOrErr(tz, "no such overload")
 		}
-		loc, err := time.LoadLocation(string(tz.(String)))
-		if err == nil {
-			return visitor(t.In(loc))
+		val := string(tz.(String))
+		ind := strings.Index(val, ":")
+		if ind == -1 {
+			loc, err := time.LoadLocation(val)
+			if err == nil {
```

Comment:
Negate the test and put a "return &Err{err}" inside the "if", and have the happy path "return visitor..." after the "if".  Then, since all paths here end in a return, you can eliminate the "else" and outdent its code.